lib/connection/pool.js

```diff
@@ -700,87 +683,73 @@ Pool.prototype.connect = function() {
     throw new MongoError('connection in unlawful state ' + this.state);
   }
 
-  var self = this;
-  // Transition to connecting state
+  const self = this;
```

Comment:
If we're replacing `self` elsewhere, can we use a more descriptive name here too?

Reply:
Okay, that works for me!